documents/model.go

```diff
@@ -754,3 +755,154 @@ func isNFTInRole(role *coredocumentpb.Role, registry common.Address, tokenID []b
 
 	return false
 }
+
+// assembleAccessToken assembles a Read Access Token from the payload received
+func assembleAccessToken(dm *CoreDocumentModel, id identity.IDConfig, payload documentpb.AccessTokenParams) (*coredocumentpb.AccessToken, error) {
+	tokenIdentifier := utils.RandomSlice(32)
+	granterID := id.ID[:]
+	roleID := byte(len(dm.Document.Roles))
+	grantee, err := hexutil.Decode(payload.GetGrantee())
+	if err != nil {
+		return nil, err
+	}
+	docID, err := hexutil.Decode(payload.GetDocumentIdentifier())
+	if err != nil {
+		return nil, err
+	}
+	// assemble access token message to be signed
+	tm := append(tokenIdentifier, granterID...)
+	tm = append(tm, grantee...)
+	tm = append(tm, roleID)
+	tm = append(tm, docID...)
+
+	// fetch key pair from identity
+	privateKey := id.Keys[identity.KeyPurposeEthMsgAuth].PrivateKey
+	pubKey := id.Keys[identity.KeyPurposeEthMsgAuth].PublicKey
+
+	// sign the assembled access token message, return signature
+	sig, err := crypto.SignMessage(privateKey, tm, "CurveSecp256K1", true)
+	if err != nil {
+		return nil, err
+	}
+	// assemble the access token, appending the signature and public keys
+	at := new(coredocumentpb.AccessToken)
```

Comment:
Why not use this construction:
```
at := coredocumentpb.AccessToken{
    Identifier: tokenIdentifier,
    Granter: granterId,
    Grantee: grantee,
    ...
}
```

Reply:
lets define constructor functions, I'm sure they would be useful to have